src/TNetwork.cpp

```diff
@@ -808,6 +808,15 @@ void TNetwork::SendFile(TClient& c, const std::string& UnsafeName) {
         return;
     }
     auto FileName = fs::path(UnsafeName).filename().string();
+
+    for (auto mod : mResourceManager.GetMods()) {
+        if (mod["file_name"].get<std::string>() == FileName && mod["protected"] == true) {
+            beammp_warn("Client tried to access protected file " + UnsafeName);
+            c.Disconnect("Mod is protected thus cannot be downloaded");
+            return;
+        }
```

Comment:
```suggestion
        const std::string& ModFileName = mod["file_name"].get<std::string>()
        if (ModFileName == FileName && mod["protected"] == true) {
            beammp_warn("Client tried to access protected file " + UnsafeName);
            c.Disconnect("The mod " + ModeFileName + " is protected, add it manually to join.");
            return;
        }
```

Feel free to reject this change if it doesn't make sense, I think it should be clear for the user why they can't join

Reply:
The client will never see this message, as the launcher will check wether a mod is protected and then disconnect and show a more detailed message to the user if it's missing.
This is more a message for when a 3rd party or malicious client tries to get a protected mod.